tf2_ros/include/tf2_ros/buffer.h

```diff
@@ -69,11 +74,43 @@ class Buffer : public BufferInterface, public AsyncBufferInterface, public tf2::
    * \param clock A clock to use for time and sleeping
    * \param cache_time How long to keep a history of transforms
    * \param node If passed advertise the view_frames service that exposes debugging information from the buffer
+   * \param  qos If passed change the quality of service of the frames_server_ service
    */
-  TF2_ROS_PUBLIC Buffer(
```

Comment:
So TF2_ROS_PUBLIC is essentially 
```
#define TF2_ROS_EXPORT __declspec(dllexport)
```

or some similar form of `dllexport`. Thus, it cannot exist in the scope of a definition of it's application [according to the docs](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2491?view=msvc-170&redirectedfrom=MSDN). We have to immediately define the constructor for `Buffer` since it's a template. As for removing the macro, the [original CI](https://ci.ros2.org/job/ci_windows/21239/console) job has linking errors in `getFrames` and `onTimeJump` which are the only two private functions used in the constructor of `Buffer`. I assume that if we mark those functions as able to link, `Buffer` shouldn't have problems with anything else